closure/goog/useragent/useragent.js

```diff
@@ -207,8 +194,12 @@ goog.userAgent.SAFARI = goog.userAgent.WEBKIT;
  * @private
  */
 goog.userAgent.determinePlatform_ = function() {
-  var navigator = goog.userAgent.getNavigator();
-  return navigator && navigator.platform || '';
+  const nav = goog.userAgent.getNavigator();
+  if (nav === null || !('platform' in nav)) {
+    return '';
+  }
+
+  return /** @type {string} */ (navigator['platform']);
```

Comment:
Indeed, this just seems to make things more brittle.  It looks like the previous code was assuming that if `navigator.platform` was not a string, then it would be falsy - so it handled `null` and `undefiend`.  Here you're only handling one type of `undefined`.